### PR TITLE
Workaround missing desktop session in tty2 on SLED

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
- Related ticket: https://bugzilla.suse.com/show_bug.cgi?id=1167633
- Verification run:
  - suportserver: https://openqa.suse.de/tests/4034754 (deleted)
  - hawk: https://openqa.suse.de/tests/4034755 (failed, missing import)

### CODE REVIEW

**Comment by alvarocarvajald** [comment]
>   * hawk: https://openqa.suse.de/tests/4034755 (running)

This will probably fail as it's a MM job. I'll try to run an equivalent SM job in my environment with this branch and get back to you with the results.

**Comment by SergioAtSUSE** [comment]
New verification runs:

- support server: https://openqa.suse.de/tests/4034772
- hawk: https://openqa.suse.de/tests/4034773

**Comment by alvarocarvajald** [comment]
SM hawk_test with the same SLED qcow2: http://mango.suse.de/tests/2162

File: tests/ha/hawk_gui.pm [hunk 1]
```diff
@@ -49,6 +49,7 @@ sub run {
     my $docker_image = "registry.opensuse.org/home/rbranco/branches/opensuse/templates/images/tumbleweed/containers/hawk_test";
     assert_script_run("docker pull $docker_image", 240);
 
+    record_soft_failure('bsc#1167633 - No desktop session in tty2 on SLED') if is_sle('15-SP2+') && check_var('SLE_PRODUCT', 'sled');
```
Comment (alvarocarvajald) [inline tests/ha/hawk_gui.pm]: Don't like too much soft failing without verifying the condition, but I guess we can add that at a later date.

Comment (okurz) [reply tests/ha/hawk_gui.pm]: We can also add the record_soft_failure later :)

Please change to not include an unconditional record_soft_failure

**Comment by alvarocarvajald** [comment]
> SM hawk_test with the same SLED qcow2: http://mango.suse.de/tests/2162

Failed due to `is_sle` not imported into `ha/hawk_gui`.

New verification run: http://mango.suse.de/tests/2163

**Comment by SergioAtSUSE** [comment]
@alvarocarvajald, can you try again?

**Comment by alvarocarvajald** [comment]
> @alvarocarvajald, can you try again?

I did. It's working: http://mango.suse.de/tests/2163

(Disregard failure while testing hawk)

**Comment by alvarocarvajald** [comment]
I still think though that it's better to revert #9659 . Don't know if you saw my comment in #9659 , but I think even in your SLES verification gdm is in tty7 and not tty2.

**Comment by SergioAtSUSE** [comment]
> I still think though that it's better to revert #9659 . Don't know if you saw my comment in #9659 , but I think even in your SLES verification gdm is in tty7 and not tty2.

GDM is always in tty7.
Desktop session is on tty2, that's the problem.

You can try on your machine (Leap 15.2+).
Log in to graphical session, then press CTRL-ALT-F7 and you always get the GDM. Then you "log in" again, and you get redirected to tty2.

**Comment by alvarocarvajald** [comment]
> GDM is always in tty7.
> Desktop session is on tty2, that's the problem.
> 
> You can try on your machine (Leap 15.2+).
> Log in to graphical session, then press CTRL-ALT-F7 and you always get the GDM. Then you "log in" again, and you get redirected to tty2.

Yeah, just checked with SLES4SAP 15-SP2. You're correct.

**Pull request merged**